test/keras-models.txt

```diff
@@ -1,7 +1,32 @@
+# Keras models from examples directory that will be used for testing
+#
+# Synthax:
+#    MODEL_NAME[:WEIGHTS_FILE] [x:XILINXPART] [c:CLOCK_PERIOD] [io:s] [r:REUSE_FACTOR] [t:AP_TYPE]
+# where
+#    MODEL_NAME - Name of the file containing json model (without ".json")
+#    WEIGHTS_FILE - Name of the HDF5 file containing model weights (without ".h5")
+#    x:XILINXPART - Xilinx part number to use
+#    c:CLOCK_PERIOD - Clock period
+#    io:s - User serial I/O, otherwise use parallel I/O
+#    r:REUSE_FACTOR - Reuse factor
+#    t:AP_TYPE - Default precision
+#
+# Lines starting with "#" are ignored.
+#
+
 KERAS_1layer
 KERAS_3layer
-KERAS_3layer:KERAS_3layer_70pruned_retrained_weights
-KERAS_conv1d
+#KERAS_3layer:KERAS_3layer_70pruned_retrained_weights
+#KERAS_conv1d
 KERAS_conv1d_small
 KERAS_conv2d_model
 KERAS_dense_big
+KERAS_3layer_batch_norm
```

Comment:
The 3layer_batch norm failed for me with the typical failure case:

```
ERROR: [XFORM 203-103] Array 'mult.V' (/home/ejk/work/HLS4ML/nnet_utils/nnet_layer.h:56): partitioned elements number (2048) has exeeded the threshold (1024), which may cause long run-time.
ERROR: [HLS 200-70] Pre-synthesis failed.
command 'ap_source' returned error code
    while executing
"source [lindex $::argv 1] "
    ("uplevel" body line 1)
    invoked from within
"uplevel \#0 { source [lindex $::argv 1] } "
```

I used Vivado 2017.4. Is this maybe a version issue? Or, does something need to include the directive in the test process that bumps up the max number of elements?

Reply:
Which branches are being compared?  

I thought we had that directive in the tcl:
https://github.com/hls-fpga-machine-learning/hls4ml/blob/master/hls-template/build_prj.tcl#L24

Reply:
Yeah, I think this directive went in after this branch was made so that makes sense.

@benjaminkreis was just telling me yesterday that `config_array_partition -maximum_size 4096` doesn't work in 2017.2 so I guess we are "supporting" 2018.1 and above...

Reply:
I like the idea of supporting older versions if it's only a matter of changing that line to that!

Reply:
With this change and with 2017.2 the 3layer_batch norm model synthesizes. I can submit a PR, but this change only prevents the build script from failing on that particular line. It doesn't solve the underlying problem that we cannot increase the maximum partition size on older versions of HLS and the issue will probably resurface with some future models. Still want it?

Reply:
Yes, I think we still want it.  The only goal of that line is to set the max above our own threshold for our "sublayer" approach.  With older versions, we won't be setting a max, but the "sublayer" approach will kick in at 4096 and prevent memory issues.  With newer versions, the max will increase to 4096, and again the sublayer approach kicks in at 4096 to prevent memory issues.  Chatted with @nhanvtran, and he agrees.